src/main/java/net/minecraftforge/client/model/generators/BlockModelBuilder.java

```diff
@@ -16,7 +16,12 @@
  *
  * @see ModelProvider
  * @see ModelBuilder
+ *
+ * In 1.21.4 Mojang exposed their data generators for their models. So it should be feasible to just use theirs.
+ * If you find something lacking feel free to open a PR so that we can extend it.
+ * @deprecated Use Vanilla's providers {@link net.minecraft.client.data.models.ModelProvider}
  */
+@Deprecated(since = "1.21.4", forRemoval = true)
```

Comment:
I can look through that myself. I've often times needed to write extensions to Forge's own model generators for my older projects. If it's not cumbersome to maintain, I'll send them your way with another PR sometime after this one is merged.